posthog/temporal/tests/data_imports/test_mysql_source.py

```diff
@@ -0,0 +1,167 @@
+"""Tests for the MySQL source.
+
+NOTE: These tests require a MySQL server to be running locally (or somewhere else).
+Therefore these tests will only run if the required environment variables are set.
+
+You can run a local MySQL server using Docker:
+
+```
+docker run -d --name mysql-test -e MYSQL_ROOT_PASSWORD=root -e MYSQL_DATABASE=test -p 3306:3306 mysql:9.2
+```
+
+Then you can run these tests using:
+
+```
+OBJECT_STORAGE_ENDPOINT=http://localhost:19000 \
+    MYSQL_HOST=localhost \
+    MYSQL_USER=root \
+    MYSQL_PASSWORD=root \
+    MYSQL_DATABASE=test \
+    pytest posthog/temporal/tests/data_imports/test_mysql_source.py
+```
+
+"""
+
+import datetime as dt
+import os
+import uuid
+
+import pymysql
+import pytest
+
+from posthog.temporal.tests.data_imports.conftest import run_external_data_job_workflow
+from posthog.warehouse.models import ExternalDataSchema, ExternalDataSource
+
+pytestmark = pytest.mark.usefixtures("minio_client")
+
+REQUIRED_ENV_VARS = (
+    "MYSQL_HOST",
+    "MYSQL_USER",
+    "MYSQL_PASSWORD",
+    "MYSQL_DATABASE",
+)
+
+MYSQL_TABLE_NAME = "test_table"
+
+TEST_DATA = [
+    (1, "John Doe", "john@example.com", dt.datetime(2025, 1, 1, tzinfo=dt.UTC), 100),
+    (2, "Jane Smith", "jane@example.com", dt.datetime(2025, 1, 2, tzinfo=dt.UTC), 2000000),
+    (3, "Bob Wilson", "bob@example.com", dt.datetime(2025, 1, 3, tzinfo=dt.UTC), 3409892966),
```

Comment:
Should we add a test here that is at the margin of what the storage capacity for these are. I found this somewhat handy reference that might be helpful. https://dev.mysql.com/doc/refman/8.4/en/integer-types.html